TestColorMap.cpp

```diff
@@ -0,0 +1,48 @@
+#include <gtest/gtest.h>
+#include <utility>
+#include <sstream>
+#include <vector>
+#include "./ColorMap.h"
+
+// Helper function to generate the expected output dynamically
+std::string generateExpectedOutput(const ColorMap& colorMap) {

```

Comment:
Isn't this replicating the implementation, along with the not-aligned column separators?
Downside is - when we fix the alignment, we have to do it in two places - test and implementation.
Can you think of a way to test the alignment, without replicating the implementation?